boa/src/object/mod.rs

```diff
@@ -74,8 +75,15 @@ pub struct Object {
 }
 
 /// Defines the different types of objects.
+#[derive(Trace, Finalize)]
+pub struct ObjectData {
+    kind: ObjectKind,
+    #[unsafe_ignore_trace]
+    internal_methods: InternalObjectMethods,
```

Comment:
```suggestion
    internal_methods: &'static InternalObjectMethods,
```

I think it's best to have the internal methods be a reference to a `static` global variable 
```rust
static STRING_INTERNAL_OBJECT_METHODS: InternalObjectMethods = { ... };
```
And we just get the reference to it, since it is a static it does not need construction at runtime and we save a lot of space because we don't store the 11 function pointers each 8 bytes (on 64-bit system), so in total we save 80 bytes of space (its minus 8 to store the `&'static InternalObjectMethods`).

Reply:
Ohhh, I didn't consider that possibility at all. Yeah, that's a great way to avoid constructing every type of internal methods at runtime. I'm gonna work in this then.